channelserver/src/meta.rs

```diff
@@ -331,6 +336,19 @@ impl From<HttpRequest<WsChannelSessionState>> for SenderData {
         // parse user-header for platform info
         sender.ua = get_ua(&headers, &log, &sender);
         get_location(&mut sender, &langs, &log, &req.state().iploc);
+
+        // If there's no sender, try pulling the GCP header.
+        // NOTE: This is US/EN only, so localization should come later.
+        if sender.city.is_none() {
+            if let Some(ghead) = headers.get("X-Client-Geo-Location") {
+                if let Ok(loc_str) = ghead.to_str() {
+                    let mut bits = loc_str.split(',').collect::<Vec<&str>>();
+                    let mut bi = bits.iter();
+                    sender.region = bi.next().map(|s| String::from(*s));
```

Comment:
nit: I'd prefer s.to_owned() just to be consistent w/ every where else

Reply:
Heh, mostly because `(*s).to_owned()` didn't look a lot better.